tests/common.go

```diff
@@ -389,31 +413,69 @@ func (ks *KubernetesSuite) waitUntilLog(ctx context.Context, namespace string, p
 		}
 		return nil
 	}
-	ks.logf("Waiting until %q pods logs "+description+": %s", ls, logMatchers)
-	mustEventually(ks.T(), ctx, checkLogs, 10*time.Second, fmt.Sprintf("Not all %q pods logs "+description, ls))
 }
 
-// containsLineMatching returns a simple line-based regex matcher to go with waitUntilLog.
-// Note: currently limited by bufio.Reader default buffer size (4KB) for simplicity.
-func containsLineMatching(re *regexp.Regexp) *lineMatcher {
-	return &lineMatcher{re: re}
+// getLastLogBytePos will return the position of the last log byte for a particular pod.  This is intended to be used with
+// the log matchers to ensure a log from previous tests isn't matched against.
+func (ks *KubernetesSuite) getLastLogBytePos(ctx context.Context, namespace, pod, container string) (int64, error) {
+	resp := ks.k8s.CoreV1().Pods(namespace).GetLogs(pod, &coreV1.PodLogOptions{Container: container}).Do(ctx)
+
+	logB, err := resp.Raw()
+	if err != nil {
+		return 0, fmt.Errorf("retrieving logs of pod %q in namespace %q failed: %w", pod, namespace, err)
+	}
+
+	return int64(len(logB)), nil
 }
 
-func allMatch(reader io.ReadSeeker, matchers ...logMatcher) (ok bool, err error) {
-	for i, matcher := range matchers {
-		_, err := reader.Seek(0, io.SeekStart)
-		if err != nil {
-			return false, fmt.Errorf("could not rewind the reader: %w", err)
-		}
-		ok, err := matcher.Match(reader)
-		if err != nil {
-			return false, fmt.Errorf("matcher %d returned an error: %w", i, err)
-		}
-		if !ok {
-			return false, nil
+// getSensorPod is a convenience method to get details of the Sensor pod.
+func (ks *KubernetesSuite) getSensorPod(ctx context.Context, namespace string) (*coreV1.Pod, error) {
+	ls := labels.SelectorFromSet(sensorPodLabels).String()
+	podList, err := ks.k8s.CoreV1().Pods(namespace).List(ctx, metaV1.ListOptions{LabelSelector: ls})
+	if err != nil {
+		return nil, fmt.Errorf("could not list pods matching %q in namespace %q: %w", sensorPodLabels, namespace, err)
+	}
+	if len(podList.Items) == 0 {
+		return nil, fmt.Errorf("empty list of pods does not satisfy the condition")
+	}
+	if len(podList.Items) > 1 {
+		return nil, fmt.Errorf("more than one sensor pod running")
+	}
+
+	return &podList.Items[0], nil
+}
+
+// waitUntilK8sDeploymentReady waits until k8s reports all pods for a deployment are ready or context done.
+func (ks *KubernetesSuite) waitUntilK8sDeploymentReady(ctx context.Context, namespace string, deploymentName string) {
+	ticker := time.NewTicker(2 * time.Second)
+	defer ticker.Stop()
+
+	timer := time.NewTimer(waitTimeout)
+
+	for {
+		select {
+		case <-ctx.Done():
+			require.NoError(ks.T(), ctx.Err())
+
+		case <-ticker.C:
+			deploy, err := ks.k8s.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metaV1.GetOptions{})
+			require.NoError(ks.T(), err, "getting deployment %q from namespace %q", deploymentName, namespace)
```

Comment:
nit: I believe you can do `ks.Require().NoErrorf(err, "getting deployment %q from namespace %q", deploymentName, namespace)`

Also, I wonder how well these instances of `ks.Require().NoError` have been working in this file. I think they've all been missing the `f`

Reply:
> missing the `f`

Is that supposed to be "missing the `t`"?

Whenever a `Run` method is executed in the suite it overwrites the require with the updated `t` - so it should be OK in representing the current testing context (ie: test A vs. sub test B)